### PR TITLE
Added feature demo 11, create-agent-with-code-interpreter

### PR DESCRIPTION
**Pull request by rostcheck**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: agents-for-bedrock/features-examples/11-create-agent-with-code-interpreter/README.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    response = client.create_agent_action_group(
+    actionGroupName='CodeInterpreterAction',
+    actionGroupState='ENABLED',
+    agentId='<YOUR_AGENT_ID>',
+    agentVersion='<YOUR_AGENT_VERSION>',
+    parentActionGroupSignature='AMAZON.CodeInterpreter'
+)
+```
+
+When invoking your agent, should supply parameter including the query text, agent id, agent alias, and a session id. Other parameters allow enabling tracing, to see the details of the model's return communication stream, ending the session, optionally storing conversational memory, and maintaining a state for this session.
```
Comment (mttanke) [inline agents-for-bedrock/features-examples/11-create-agent-with-code-interpreter/README.md:27]: there is no mention on how to use code interpretation for inference here. This is for memory, needs to be updated for code interpretation

File: agents-for-bedrock/features-examples/11-create-agent-with-code-interpreter/files/SPY.csv [hunk 2]
```diff
@@ -0,0 +1,254 @@
+Symbol,Date,Open,High,Low,Close,Adj Close,Volume
```
Comment (mttanke) [inline agents-for-bedrock/features-examples/11-create-agent-with-code-interpreter/files/SPY.csv:1]: need to make reference to data source and preferably download the data from the source on the notebook

File: agents-for-bedrock/features-examples/11-create-agent-with-code-interpreter/11-create-agent-with-code-interpreter.ipynb [hunk 3]
```diff
@@ -0,0 +1,2479 @@
...
+   "id": "c603ce06-61d8-4236-92e8-883e0d49cfea",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "\n",
```
Comment (mttanke) [inline agents-for-bedrock/features-examples/11-create-agent-with-code-interpreter/11-create-agent-with-code-interpreter.ipynb:54]: clean the notebook outputs and restart it

File: agents-for-bedrock/features-examples/11-create-agent-with-code-interpreter/11-create-agent-with-code-interpreter.ipynb [hunk 4]
```diff
@@ -0,0 +1,2479 @@
...
+   "execution_count": 7,
+   "id": "0dcb5332-c035-435a-91e5-838ec95d9a7d",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "# configuration variables\n",
+    "suffix = f\"{region}-{account_id}\"\n",
+    "agent_name = \"assistant-w-code-interpret\"\n",
```
Comment (mttanke) [inline agents-for-bedrock/features-examples/11-create-agent-with-code-interpreter/11-create-agent-with-code-interpreter.ipynb:225]: assistant-w-code-interpretation --- we need to use the marketing name

File: agents-for-bedrock/features-examples/11-create-agent-with-code-interpreter/11-create-agent-with-code-interpreter.ipynb [hunk 5]
```diff
@@ -0,0 +1,2479 @@
...
+  },
+  {
+   "cell_type": "markdown",
+   "id": "65efbf63-fbdb-4220-ab89-c7af740447ef",
+   "metadata": {},
+   "source": [
+    "### Create Agent Action Group\n",
+    "In Bedrock agents, action groups define tools for the agent to use. We will now create an agent action group to provide the agent with code interpreter, a runtime environment for evaluating code. Action groups can also define other tools, such as lambda functions, and can also define a channel for the model to solicit clarifying input from the user if needed (treating the user as a tool that the model can invoke). Our action group, however, just defines the code interpreter. This is done via a special access parameter, parentActionGroupSignature (see [boto3 documentation](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/bedrock-agent/client/create_agent_action_group.html))\n",
+    "\n",
+    "To allow your agent to generate, run, and troubleshoot code when trying to complete a task, set parentActionGroupSignature to AMAZON.CodeInterpreter. You must leave the description, apiSchema, and actionGroupExecutor fields blank for this action group.\n",
```
Comment (mttanke) [inline agents-for-bedrock/features-examples/11-create-agent-with-code-interpreter/11-create-agent-with-code-interpreter.ipynb:434]: make AMAZON.CodeInterpreter code style: `AMAZON.CodeInterpreter`

Comment (mttanke) [reply agents-for-bedrock/features-examples/11-create-agent-with-code-interpreter/11-create-agent-with-code-interpreter.ipynb:434]: (need to add ` before and after)

File: agents-for-bedrock/features-examples/11-create-agent-with-code-interpreter/11-create-agent-with-code-interpreter.ipynb [hunk 6]
```diff
@@ -0,0 +1,2479 @@
...
+    "\n",
+    "Note that you can also define an action group with parentActionGroupSignature set to the special value AMAZON.UserInput. If this is set, then during orchestration, if your agent determines that it needs to invoke an API in an action group, but doesn’t have enough information to complete the API request, it will invoke this action group instead and return an Observation reprompting the user for more information. User input is appropriate if you know the interaction has a human in the loop. We do not do that here."
```
Comment (mttanke) [inline agents-for-bedrock/features-examples/11-create-agent-with-code-interpreter/11-create-agent-with-code-interpreter.ipynb:436]: make parentActionGroupSignature and AMAZON.UserInput code style

File: agents-for-bedrock/features-examples/11-create-agent-with-code-interpreter/11-create-agent-with-code-interpreter.ipynb [hunk 7]
```diff
@@ -0,0 +1,2479 @@
...
+       "  'updatedAt': datetime.datetime(2024, 7, 14, 17, 7, 42, 141256, tzinfo=tzutc())}}"
+      ]
+     },
+     "execution_count": 14,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "response"
```
Comment (mttanke) [inline agents-for-bedrock/features-examples/11-create-agent-with-code-interpreter/11-create-agent-with-code-interpreter.ipynb:542]: remove cell with response only

File: agents-for-bedrock/features-examples/11-create-agent-with-code-interpreter/11-create-agent-with-code-interpreter.ipynb [hunk 8]
```diff
@@ -0,0 +1,2479 @@
...
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 18,
+   "id": "ce1f5e94",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def invoke_agent_helper(\n",
+    "    query, session_id, agent_id, alias_id, enable_trace=False, memory_id=None, session_state=None, end_session=False, show_code_use=False\n",
```
Comment (mttanke) [inline agents-for-bedrock/features-examples/11-create-agent-with-code-interpreter/11-create-agent-with-code-interpreter.ipynb:676]: add a break line here to make it easier to see

Comment (mttanke) [reply agents-for-bedrock/features-examples/11-create-agent-with-code-interpreter/11-create-agent-with-code-interpreter.ipynb:676]: maybe move this one one cell up to match documentation

File: agents-for-bedrock/features-examples/11-create-agent-with-code-interpreter/11-create-agent-with-code-interpreter.ipynb [hunk 9]
```diff
@@ -0,0 +1,2479 @@
...
+    "    return process_response(agent_response, enable_trace=enable_trace, show_code_use=show_code_use)\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "86f6763d-220d-49e2-b08e-9cf5a0494128",
+   "metadata": {},
+   "source": [
+    "## Invoking code interpreter\n",
+    "We ask the agent to generate a random string, which will require it to use the code interpreter. Using the show_code_use flag, we can see that the agent invokes code interpreter to evaluate the python code it generates."
```
Comment (mttanke) [inline agents-for-bedrock/features-examples/11-create-agent-with-code-interpreter/11-create-agent-with-code-interpreter.ipynb:701]: make show_code_use code style

File: agents-for-bedrock/features-examples/11-create-agent-with-code-interpreter/11-create-agent-with-code-interpreter.ipynb [hunk 10]
```diff
@@ -0,0 +1,2479 @@
...
+    "    return session_state"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "666c49cf",
+   "metadata": {},
+   "source": [
+    "### Passing files for normal chat\n",
+    "Here we pass in a local CSV file and ask the model to explain what the data is. Note that when adding the file to the session state, we specify use case 'CHAT' instead of 'CODE_INTERPRETER' and by setting show_code_use=True for our helper we see that the model does not use the code interpreter, it assesses the information using the LLM model's intelligence."
```
Comment (mttanke) [inline agents-for-bedrock/features-examples/11-create-agent-with-code-interpreter/11-create-agent-with-code-interpreter.ipynb:888]: make 'CHAT' and 'CODE_INTERPRETER' code style

File: agents-for-bedrock/features-examples/11-create-agent-with-code-interpreter/11-create-agent-with-code-interpreter.ipynb [hunk 11]
```diff
@@ -0,0 +1,2479 @@
...
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "936efbf9",
+   "metadata": {},
+   "source": [
+    "First, we examine the file ourselves. We see it is a list of historical prices for the S&P 500 index."
```
Comment (mttanke) [inline agents-for-bedrock/features-examples/11-create-agent-with-code-interpreter/11-create-agent-with-code-interpreter.ipynb:896]: Add reference to the data and download it from source

**Review by mttanke — CHANGES_REQUESTED** [review]

File: agents-for-bedrock/features-examples/11-create-agent-with-code-interpreter/README.md (hunk 1 above)
Comment (rostcheck) [reply agents-for-bedrock/features-examples/11-create-agent-with-code-interpreter/README.md:27]: The code interpretation doesn't actually need any parameters, if it's enabled via the special action group, the agent just decides if it wants to use it. I could add a note about that...

File: agents-for-bedrock/features-examples/11-create-agent-with-code-interpreter/11-create-agent-with-code-interpreter.ipynb (hunk 4 above)
Comment (rostcheck) [reply agents-for-bedrock/features-examples/11-create-agent-with-code-interpreter/11-create-agent-with-code-interpreter.ipynb:225]: Ah, it's too long, that's why I shortened it...

**Review by mttanke — APPROVED** [review]

**Pull request merged**